### ISSUE TITLE
This is my Config.ini controls for The Last of Us (and some other games).

### ISSUE THREAD

**Issue by idgas01** [r57zone/DualShock4-emulator#11]
### Game ► The Last of Us

DS4Emulator 1.75

```
# THE LAST OF US

# L1 --- RMB for "Zoom".
L1=2

# R1 --- LMB for "Fire, Reload Weapon, Use Health".
R1=1

# L2 --- Shift for "Sprint".
L2=16

# R2 --- Ctrl for "Listen Mode, Switch Shoulders While Aiming".
R2=17

# L3 --- F for "Focus On Hint/Scene".
L3=70

# R3 --- Alt for "Flashlight".
R3=18

# ------ R for "Recharge Flashlight".
SHAKING=82

# O ---- C for "Crouch, Previous Menu".
CIRCLE=67

# X ---- Space for "Menu Select".
CROSS=32

# △ ---- E for "Interact".
TRIANGLE=69

# □ ---- Q for "Melee, Escaping, Other".
SQUARE=81

# ------ Tab for "Backpack/Crafting".
SHARE=9

# ------ O for "Options".
OPTIONS=79

# ------ Up, Down, Left, Right for "Weapons/Items Menu".
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ W, S, A, D.
LS_UP=87
LS_DOWN=83
LS_LEFT=65
LS_RIGHT=68

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  9 for "Swipe Left".
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  0 for "Swipe Right".
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  7 for "Swipe Up".
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  8 for "Swipe Down".
TOUCHPAD_SWIPE_DOWN=56

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  H for "Touchpad Left".
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  K for "Touchpad Right".
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  J for "Touchpad Center".
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  U for "Touchpad Up".
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  N for "Touchpad Down".
TOUCHPAD_DOWN=78

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  Enter for "Touchpad".
TOUCHPAD=13

# ------ F10 for the PlayStation Button (Return to Game, Close Game, Application Saved Data Management, Vibration Settings).
PS=121
```

In DS4Emulator 1.7.5, there is a button for "Motion" that lets you shake your controller.‎  This should allow you to recharge your flashlight.  I haven't tested it out yet though.  If it doesn't work, another way to recharge your flashlight is to just restart the checkpoint.

### Game ► The Last of Us Part II

DS4Emulator 1.75

```
[Keys]

# THE LAST OF US PART II

# L1 --- Shift for "Sprint, Previous Menu".
L1=16

# R1 --- Ctrl for "Listen Mode, Switch Weapons, Next Menu".
R1=17

# L2 --- RMB for "Zoom".
L2=2

# R2 --- LMB for "Fire, Reload Weapon, Use Health".
R2=1

# L3 --- F for "Focus On Hint/Scene".
L3=70

# R3 --- Alt for "Flashlight".
R3=18

# O ---- C for "Crouch, Previous Menu".
CIRCLE=67

# X ---- Space for "Jump, Climb, Menu Select".
CROSS=32

# △ ---- E for "Interact".
TRIANGLE=69

# □ ---- Q for "Melee, Escaping, Switch Shoulders While Aiming".
SQUARE=81

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  Enter for "".
SHARE=13

# ------ Tab for Backpack/Crafting/Journal".
TOUCHPAD=9

# ------ O for "Options".
OPTIONS=79

# ------ Up, Down, Left, Right for "Weapons/Items Menu".
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ W, S, A, D.
LS_UP=87
LS_DOWN=83
LS_LEFT=65
LS_RIGHT=68

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  9 for "Swipe Left".
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  0 for "Swipe Right".
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  7 for "Swipe Up".
TOUCHPAD_SWIPE_UP=55
# ------ X for "Play Guitar Note".
TOUCHPAD_SWIPE_DOWN=88

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  H for "Touchpad Left".
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  K for "Touchpad Right".
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  J for "Touchpad Center".
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  U for "Touchpad Up".
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  N for "Touchpad Down".
TOUCHPAD_DOWN=78

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  R for "Motion".
SHAKING=82

# ------ F10 for the PlayStation Button (Return to Game, Close Game, Application Saved Data Management, Vibration Settings).
PS=121
```

‎
### Game ► Until Dawn

DS4Emulator 1.6.  I will update this to 1.75 later on.

```
[Keys]

# UNTIL DAWN

# L1 --- RMB for "Next Tab in Character Menu, Walk Faster"
L1=2

# R1 --- LMB for "Opens the Character Menu, Previous Tab in Character Menu"
R1=1

# L2 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Shift)
L2=16

# R2 --- Ctrl for "Grab Objects, Fire Weapons"
R2=17

# L3 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (F)
L3=70

# R3 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Alt)
R3=18

# O ---- C for "Button Prompt, Previous Menu"
CIRCLE=67

# △ ---- E for "Button Prompt"
TRIANGLE=69

# □ ---- Q for "Button Prompt"
SQUARE=81

# X ---- Space for "Interact, Menu Select"
CROSS=32

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Tab)
SHARE=9

# ------ O for "Options"
OPTIONS=79

# ------ Up, Down, Left, Right
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ 9  (If it doesn't work on the first try, keep pressing.  It will eventually work.)
TOUCHPAD_SWIPE_LEFT=57
# ------ 0  (If it doesn't work on the first try, keep pressing.  It will eventually work.)
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (7)
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (8)
TOUCHPAD_SWIPE_DOWN=56

# ------ H  (If you press K and then H, it is the same thing as a Swipe Left.)
TOUCHPAD_LEFT=72
# ------ K  (If you press H and then K, it is the same thing as a Swipe Right.)
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (J)
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (U)
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (N)
TOUCHPAD_DOWN=78
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Enter)
TOUCHPAD=13
```

### Game ► Little Nightmares

DS4Emulator 1.6.  I will update this to 1.75 later on.

```
[Keys]

# LITTLE NIGHTMARES

# L1 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (RMB)
L1=2

# R1 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (LMB)
R1=1

# L2 --- C for "Crouch, Slide"
L2=67

# R2 --- F for "Grab"
R2=70

# L3 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (1)
L3=49

# R3 --- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (2)
R3=50

# O ---- Alt for "Lighter, Previous Menu"
CIRCLE=18

# X ---- Space for "Jump, Throw"
CROSS=32

# △ ---- YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (3)
TRIANGLE=51

# □ ---- Shift for "Sprint"
SQUARE=16

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Tab)
SHARE=9

# ------ O for "Options"
OPTIONS=79

# ------ Up, Down, Left, Right
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (9)
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (0)
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (7)
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (8)
TOUCHPAD_SWIPE_DOWN=56

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (H)
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (K)
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (J)
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (U)
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (N)
TOUCHPAD_DOWN=78
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Enter)
TOUCHPAD=13
```

### Game ► The Darkness

DS4Emulator 1.6.  I will update this to 1.75 later on.

```
[Keys]

# THE DARKNESS

# SPAWN DARKLINGS:
# Press Ctrl + F (□) to spawn a Berserker Darkling.
# Press Ctrl + E (X) to spawn a Gunner Darkling.
# Press Ctrl + Q (O) to spawn a Kamikaze Darkling.
# Press Ctrl + Space (△) to spawn a Lightkiller Darkling.

# L1 --- Ctrl for "Manifest Darkness"
L1=17

# R1 --- Shift for "Darkness Powers: Creeping Dark, Demon Arm, Darkness Guns, Black Hole"
R1=16

# L2 --- LMB for "Fire Left Weapon"
L2=1

# R2 --- RMB for "Fire Right Weapon"
R2=2

# L3 --- C for "Crouch"
L3=67

# R3 --- Alt for "Zoom"
R3=18

# O ---- Q for "Reload Weapon, Previous Menu"
CIRCLE=81

# X ---- E for "Interact, Devour Hearts, Start Button, Menu Select"
CROSS=69

# △ ---- Space for "Jump"
TRIANGLE=32

# □ ---- F for "Redirect Darklings"
SQUARE=70

# ------ Tab for "Objectives"
SHARE=9

# ------ O for "Options"
OPTIONS=79

# ------ Switch Darkness Powers
DPAD_UP=38
DPAD_DOWN=40
# ------ Switch Weapons
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (9)
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (0)
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (7)
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (8)
TOUCHPAD_SWIPE_DOWN=56

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (H)
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (K)
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (J)
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (U)
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (N)
TOUCHPAD_DOWN=78
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Enter)
TOUCHPAD=13
```

### Game ► Red Dead Redemption

DS4Emulator 1.6.  I will update this to 1.75 later on.

```
[Keys]

# RED DEAD REDEMPTION

# I just started playing, so I might not know all of the controls yet.

# L1 --- Q for "Weapons Menu"
L1=81

# R1 --- Ctrl for "Take Cover"
R1=17

# L2 --- RMB for "Draw Weapon, Aim Weapon"
L2=2

# R2 --- LMB for "Use Weapon"
R2=1

# L3 --- C for "Crouch"
L3=67

# R3 --- F for "Dead Eye, Change View"
R3=70

# O ---- R for "Reload, Previous Menu"
CIRCLE=82

# X ---- Alt for "Sprint, Menu Option"
CROSS=18

# △ ---- E for "Interact, Slide Down"
TRIANGLE=69

# □ ---- Space for "Jump"
SQUARE=32

# ------ Tab for "Inventory"
SHARE=9

# ------ O for "Start, Main Menu"
OPTIONS=79

# ------ Up, Down, Left, Right
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (9)
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (0)
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (7)
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (8)
TOUCHPAD_SWIPE_DOWN=56

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (H)
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (K)
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (J)
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (U)
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (N)
TOUCHPAD_DOWN=78
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Enter)
TOUCHPAD=13
```

### Game ► Uncharted: Drake's Fortune

DS4Emulator 1.6.  I will update this to 1.75 later on.

```
[Keys]

# UNCHARTERED 1

# I just started playing, so I might not know all of the controls yet.

# L1 --- RMB for "Aim Weapon"
L1=2

# R1 --- LMB for "Fire Weapon"
R1=1

# L2 --- Alt for "Zoom In"
L2=18

# R2 --- R for "Reload"
R2=82

# L3 --- YOU MIGHT NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (F)
L3=70

# R3 --- YOU MIGHT NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Shift)
R3=16

# O ---- C for "Roll, Take Cover, Previous Menu"
CIRCLE=67

# X ---- Space for "Menu Select"
CROSS=32

# △ ---- E for "Collect Ammo"
TRIANGLE=69

# □ ---- Q for "Punch"
SQUARE=81

# ------ YOU MIGHT NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Tab)
SHARE=9

# ------ O for "Options"
OPTIONS=79

# ------ Up, Down, Left, Right
DPAD_UP=38
DPAD_DOWN=40
DPAD_LEFT=37
DPAD_RIGHT=39

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (9)
TOUCHPAD_SWIPE_LEFT=57
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (0)
TOUCHPAD_SWIPE_RIGHT=48
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (7)
TOUCHPAD_SWIPE_UP=55
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (8)
TOUCHPAD_SWIPE_DOWN=56

# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (H)
TOUCHPAD_LEFT=72
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (K)
TOUCHPAD_RIGHT=75
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (J)
TOUCHPAD_CENTER=74
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (U)
TOUCHPAD_UP=85
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (N)
TOUCHPAD_DOWN=78
# ------ YOU DO NOT NEED THIS BUTTON FOR THIS GAME.  THIS IS JUST A PLACEHOLDER. (Enter)
TOUCHPAD=13
```

### Button Key Codes

This is a commented out section that I added to the bottom of my DS4Emulator config file.  It contains a list of all the button key codes.  I thought that it would be more convenient to have all of the button key codes in my config file instead of having to use my browser to visit a website https://github.com/r57zone/Half-Life-Alyx-novr/blob/master/BINDINGS.md

I changed the order of the button codes according to my personal preferences. And I changed the order of a few key names.  For example, I changed the key names "`, <`" and "`. >`" to "`< ,`" and "`> .`"

Please let me know if I made any mistakes or if any of these key codes quit working in the future.

```
# THE FOLLOWING IS A LIST OF ALL THE BUTTON CODES.

#    1 - Mouse Left Button
#    2 - Mouse Right Button
#    4 - Mouse Middle Button
#    5 - Mouse Side Button (1)
#    6 - Mouse Side Button (2)

#   16 - Shift
#   17 - Ctrl
#   18 - Alt
#   32 - Space
#    9 - Tab
#   13 - Enter
#   20 - CapsLock
#    8 - Backspace

#   65 - A a
#   66 - B b
#   67 - C c
#   68 - D d
#   69 - E e
#   70 - F f
#   71 - G g
#   72 - H h
#   73 - I i
#   74 - J j
#   75 - K k
#   76 - L l
#   77 - M m
#   78 - N n
#   79 - O o
#   80 - P p
#   81 - Q q
#   82 - R r
#   83 - S s
#   84 - T t
#   85 - U u
#   86 - V v
#   87 - W w
#   88 - X x
#   89 - Y y
#   90 - Z z

#   49 - 1 !
#   50 - 2 @
#   51 - 3 #
#   52 - 4 $
#   53 - 5 %
#   54 - 6 ^
#   55 - 7 &
#   56 - 8 *
#   57 - 9 (
#   48 - 0 )

#  112 - F1
#  113 - F2
#  114 - F3
#  115 - F4
#  116 - F5
#  117 - F6
#  118 - F7
#  119 - F8
#  120 - F9
#  121 - F10
#  122 - F11
#  123 - F12

#  192 - ~
#  189 - -
#  187 - + =
#  219 - {
#  221 - {
#  188 - < ,
#  190 - > .
#  191 - / ?
#  220 - \
#  186 - ; :
#  222 - ' "

#   96 - Numpad 0
#   97 - Numpad 1
#   98 - Numpad 2
#   99 - Numpad 3
#  100 - Numpad 4
#  101 - Numpad 5
#  102 - Numpad 6
#  103 - Numpad 7
#  104 - Numpad 8
#  105 - Numpad 9
#  109 - Numpad - (minus)
#  107 - Numpad + (plus)
#  111 - Numpad / (divide)
#  106 - Numpad * (multiply)
#  110 - Numpad Del

#   27 - Esc
#   45 - Insert
#   33 - PageUp
#   34 - PageDown
#   35 - End
#   36 - Home
#   46 - Delete
#   44 - PrintScreen
#  145 - ScrollLock
#   91 - LWin
#   92 - RWin
#   93 - Menu

#   38 - Up Arrow
#   40 - Down Arrow
#   37 - Left Arrow
#   39 - Right Arrow
```

### UTF-8 vs ANSI Encoding

In my config file, I use symbols like `△` and `□` instead of `/\` and `[]`.  But if you copy and paste those `△` and `□` symbols into the default config file, it won't let you save the file without losing those symbols.  Why?  When people who use Windows XP, Vista, 7, or 8.1 create a new Notepad document and then save it, the default encoding will be ANSI which doesn't support these symbols.  The DS4Emulator developer is using this ANSI encoding in his config files.  This type of ANSI encoding is outdated.  In Windows 10, the default encoding for all Notepad documents is UTF-8, and that's great because that means it supports these special symbols.  It is very easy to create a Notepad document in Windows XP, Vista, 7, 8.1, and 10 that uses UTF-8 instead of ANSI encoding.  Just create a new Notepad document, click 'Save As', and then next to the Save button is a drop down menu where you can change the encoding to UTF-8.  Once you save the file, you can rename it to "Config", and then you can copy and paste the contents of your original config file into it.  Then change the file extension of the config file to .ini.

### Hiding the Mouse Cursor

The DS4Emulator program lets you hide the mouse cursor.  In version 1.75, this is accomplished by changing HideCursorAfterStart=0 to HideCursorAfterStart=1.  In version 1.76, you will need to set a hotkey to hide the cursor.  But this DS4Emulator feature is very flawed.  Once the mouse cursor is hidden, you can't get it back until you completely exit the DS4Emulator program.  This is very inconvenient to people who switch tabs often and want to be able to see their cursor to move it around and type things.

That is why I am suggesting to use my program instead.  Actually, it isn't my program.  It is an AutoHotKeys script that someone created ages ago, and I just changed the name of the program and changed the hotkey.  The hotkey that I used is Win + C to toggle mouse cursor off and on.  Click [**Win+C Toggles Cursor Visibility**](https://mega.nz/file/Rw5ySB4S#_b2DPhqnfTcS4sN3NtWhc0a-z9QODe3_5XfUgP98hms) to download it.  This is the original AutoHotKey script [https://autohotkey.com/board/topic/22037-hide-and-show-mouse-cursor-by-macro](https://autohotkey.com/board/topic/22037-hide-and-show-mouse-cursor-by-macro).  Some reasons why my program is good:
1.) It does not take up any taskbar icon space and it does not create any additional command prompt or other program windows on your screen.  You will only see the program icon in the system tray.
2.) It's very simple to hide and show the cursor.  It doesn't matter what window you are in.  Just press Win + C to toggle the mouse visibility off and on. as as many times as you want.  You will never have to switch to a command prompt window, close the program, or open up the program again.  In fact, you can just create a shortcut of this program and put it in your Startup folder and leave the program running 24/7, 365 days a year.

The DS4Emulator developer also had a program called [HideCursorTool](https://github.com/r57zone/DualShock4-emulator/releases/download/1.6/HideCursorTool.zip) that lets you hide the mouse cursor.  This is a bad program for a lot of reasons.
1.) It opens up a command prompt window which takes up space on your taskbar.  And this window also takes up space on your desktop screen.
2.) If you want to make the mouse cursor visible again, you have to do two tedious things.  The first thing that you have to do is Alt + Tab to switch to the command prompt window.  Then you have to press 'Enter' which shuts down the HideCursorTool program.
3.) If you want to make the mouse cursor hidden again, then what?  You have to find the HideCursorTool exe in your File Explorer (or Desktop, or Start Menu) and then start it up again.  That's not fun.  It takes up too much time, especially if you are constantly Alt + Tab back and forth between windows and need to see and use your mouse a lot when you are in those other windows.
4.) I tested this program out and discovered that it is glitchy.  It made my mouse cursor permanently disappear and I couldn't get it back because the HideCursorTool command prompt window had already been closed.  I was forced to 'Sign Out' of my computer and sign back in to restore my mouse cursor.

**Comment by idgas01** [comment]
![psnowlauncher_VQrSl3yneu](https://user-images.githubusercontent.com/75463665/108476772-0dd9d400-7258-11eb-8ec2-97111e5f891d.jpg)

I was stuck here for a few minutes, not knowing what to do.  You are supposed to use the L stick on your controller.  I tried many different things.  Finally, what worked for me was that I spammed the WASD keys (using the above controls) while continuously left-mouse clicking.

Just wanted to make a note of this in case someone plays this game and uses this awesome emulator.
